processing/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -308,26 +355,154 @@ private void takeSnapshot(Map<String, LookupExtractorFactoryContainer> lookupMap
     }
   }
 
-  private void loadSnapshotAndInitStateRef()
+  private void loadAllLookupsAndInitStateRef()
   {
-    if (lookupSnapshotTaker != null) {
-      ImmutableMap.Builder<String, LookupExtractorFactoryContainer> builder = ImmutableMap.builder();
-
-      final List<LookupBean> lookupBeanList = lookupSnapshotTaker.pullExistingSnapshot();
-      for (LookupBean lookupBean : lookupBeanList) {
-        LookupExtractorFactoryContainer container = lookupBean.getContainer();
+    if (!lookupConfig.getDisableLookupSync()) {
+      String tier = lookupListeningAnnouncerConfig.getLookupTier();
+      List<LookupBean> lookupBeanList = new ArrayList<>();
+      // Check if the coordinator is accessible
+      if (getCoordinatorUrl().isEmpty()) {
+        if (lookupSnapshotTaker != null) {
+          LOG.info("Coordinator is unavailable. Loading saved snapshot instead");
+          lookupBeanList = getLookupListFromSnapshot();
+        }
+      } else {
+        lookupBeanList = getLookupListFromCoordinator(tier);
+        if (lookupBeanList == null) {
+          LOG.info("Lookups not accessible from Coordinator. Loading saved snapshot instead");
+          lookupBeanList = getLookupListFromSnapshot();
+        }
+      }
+      if (!lookupBeanList.isEmpty()) {
+        ImmutableMap.Builder<String, LookupExtractorFactoryContainer> builder = ImmutableMap.builder();
+        ListeningScheduledExecutorService executorService = MoreExecutors.listeningDecorator(
+            Executors.newScheduledThreadPool(
+                lookupConfig.getNumLookupLoadingThreads(),
+                Execs.makeThreadFactory("LookupReferencesManager--%s")
```

Comment:
can we name it different from the other long running .e.g LookupReferencesManager-startup-%s ?